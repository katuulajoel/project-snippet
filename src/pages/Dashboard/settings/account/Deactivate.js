/* eslint-disable react/no-unescaped-entities */
/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import React from "react";
import PropTypes from "prop-types";
/* ------------------------- Component dependencies ------------------------- */
import Button from "../../../../components/Button";

const ChangeEmail = ({ deactivate, page }) => {
  return (
    <form onSubmit={deactivate} aria-label="deactivate-form" id="deactivate">
      <div className="header">Deactivate account</div>
      <div>
        <Button
          className="save"
          type="submit"
          disabled={page && page.saving === "deactivate"}
        >
          {page && page.saving === "deactivate"
            ? "Deactivating your account ..."
            : "Deactivate my account"}
        </Button>
      </div>
    </form>
  );
};

ChangeEmail.propTypes = {
  deactivate: PropTypes.func,
  page: PropTypes.object,
};

export default ChangeEmail;
