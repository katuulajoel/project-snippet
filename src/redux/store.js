import { applyMiddleware, createStore } from "redux";
import reduxThunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

import monitorReducersEnhancer from "../configs/enhancers/monitorReducer";
import loggerMiddleware from "../configs/middleware/logger";
import rootReducer from "./reducers";

export function configureStore(preloadedState = {}) {
  const middlewares = [loggerMiddleware, reduxThunk];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer, monitorReducersEnhancer];
  const composedEnhancers = composeWithDevTools(...enhancers);

  const store = createStore(rootReducer, preloadedState, composedEnhancers);

  if (process.env.NODE_ENV !== "production" && module.hot) {
    module.hot.accept("./reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
}

const store = configureStore();
export default store;
