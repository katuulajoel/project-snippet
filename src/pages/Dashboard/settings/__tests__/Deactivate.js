import React from "react";
import { Provider } from "react-redux";
import { cleanup, render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Deactivate from "../account/Deactivate";
import store from "../../../../redux/store";

afterEach(cleanup);
describe("Deactivate layout test", () => {
  it("Snapshot test for Deactivate component", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Deactivate />
        </Router>
      </Provider>
    );

    expect(asFragment(<Deactivate />)).toMatchSnapshot();
  });
});
