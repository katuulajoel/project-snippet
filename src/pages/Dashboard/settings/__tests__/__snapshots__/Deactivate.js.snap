// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deactivate layout test Snapshot test for Deactivate component 1`] = `
<DocumentFragment>
  <form
    aria-label="deactivate-form"
    id="deactivate"
  >
    <div
      class="header"
    >
      Deactivate account
    </div>
    <div>
      <button
        class="btn btn-default save  "
        type="submit"
      >
        Deactivate my account
      </button>
    </div>
  </form>
</DocumentFragment>
`;
